src/main/java/wromaciej/hvac_sim/thermo/streams/model/AnyStream.java

```diff
@@ -0,0 +1,23 @@
+package wromaciej.hvac_sim.thermo.streams.model;
+
+import wromaciej.hvac_sim.thermo.generals.Item;
+import wromaciej.hvac_sim.ids.IdGenerator;
+import wromaciej.hvac_sim.thermo.generals.bonds.InletStreamBond;
+import wromaciej.hvac_sim.thermo.generals.bonds.OutletStreamBond;
+
+public abstract class AnyStream extends Item {
+
+    public final InletStreamBond<? extends AnyStream> inletStreamBond;
```

Comment:
parametrise and do not let somethind like `<? extends AnyStream>` because later on you need to cast